packages/terra-clinical-item-collection/src/Item.jsx

```diff
@@ -0,0 +1,189 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import ItemView from 'terra-clinical-item-view';
+import List from 'terra-list';
+import Table from 'terra-table';
+import 'terra-base/lib/baseStyles';
+import Utils from './_ItemCollectionUtils';
+import styles from './ItemCollection.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The display elements to be presented.
+   */
+  children: PropTypes.node,
+  /**
+   * The start accessory element to be presented.
+   */
+  startAccessory: PropTypes.node,
+  /**
+   * The comment element to be presented.
+   */
+  comment: PropTypes.node,
+  /**
+   * The end accessory element to be presented.
+   */
+  endAccessory: PropTypes.node,
+  /**
+   * When displayed as a list item, the column layout in which to present the displays.
+   */
+  listItemLayout: PropTypes.oneOf(['oneColumn', 'twoColumns']),
+  /**
+   * When displayed as a list item, the text color emphasis when using the two columns layout.
+   */
+  listItemTextEmphasis: PropTypes.oneOf(['default', 'start']),
+  /**
+   * When displayed as a list item, whether or not all text should truncate.
+   */
+  isListItemTruncated: PropTypes.bool,
+  /**
+   * The vertical alignment of the start and end accessories.
+   */
+  accessoryAlignment: PropTypes.oneOf(['alignTop', 'alignCenter']),
+  /**
+   * Wether or not the item is selectable. If true, the item is given list and table hover and focus styles, tabIndex
+   * set to 0, and onClick and onKeyDown callbacks set to the onSelect function provided via Item Collection.
+   */
+  isSelectable: PropTypes.bool,
+  /*
+   * Wether or not the item is selected.
+   */
+  isSelected: PropTypes.bool,
+  /**
+   * The view in which the item should be presented. Options are list or table and this will be set by the Item Collection component.
+   */
+  view: PropTypes.oneOf(['list', 'table']),
+};
+
+const defaultProps = {
+  startAccessory: undefined,
+  comment: undefined,
+  endAccessory: undefined,
+  listItemLayout: 'oneColumn',
+  listItemTextEmphasis: 'default',
+  isListItemTruncated: false,
+  accessoryAlignment: 'alignCenter',
+  isSelectable: false,
+  isSelected: false,
+  view: 'list',
+};
+
+function createSelectableProps(onSelect, itemKey) {
+  const selectableProps = { isSelectable: true };
+  selectableProps.tabIndex = 0;
+
+  if (!itemKey) {
+    /* eslint-disable no-console */
+    console.error('Key is required for correct selectable implementation.');
```

Comment:
seems like we should set and return isSelectable: false and tabIndex: -1 here right away then if no key and it is required for correct implementation

Reply:
This only throws an error & will still create the onClick and onKeyDown events for the item. However the key is the callback for these function, so if key isn't specified when the ItemCollection was created, this will return null.